Resolves #3560 - Missing captcha v2 in "error_nopermission"

Ben-MyBB opened the pull request:
Note: This PR removes the login form from the error_nopermission template and leaves a single "Login" button which takes you to the full login form.

Resolves #3560

JordanMussi commented on install/resources/mybb_theme.xml:
```diff
@@ -3639,36 +3639,15 @@ var announcement_quickdelete_confirm = "{$lang->announcement_quickdelete_confirm
-		<template name="error_nopermission" version="1818"><![CDATA[{$lang->error_nopermission_guest_1}
+		<template name="error_nopermission" version="1824"><![CDATA[{$lang->error_nopermission_guest_1}
 <ol>
 <li>{$lang->error_nopermission_guest_2}</li>
```
```suggestion
<li>{$lang->error_nopermission_guest_2} <a href="member.php?action=login">{$lang->login}</a> | <a href="member.php?action=register">{$lang->need_reg}</a></li>
```

I think it's best to add some links here and change the language line:
https://github.com/mybb/mybb/blob/36e8e7167c693ad0ac78dd05406e4cfa8fe0a9eb/inc/languages/english/messages.lang.php#L54

Ben-MyBB commented:
Thanks @JordanMussi . I have updated the PR. I'm wondering if the links we have added are actually sufficent, the buttons look out of place imo and the links fit much better. 

The buttons work fine on my local host, and action is set. If we decide to keep the buttons I can look into it further tomorrow.

Ben-MyBB commented:
@JordanMussi any thoughts?

JordanMussi commented:
Yeah I think the links inline are better than the text saying something like "use the buttons at the bottom of the page" :-)

Irrelevant if we are dropping the buttons but odd that it didn't work for me when testing because I needed a `<input type="hidden" name="action" value="login">` to get it working.

Ben-MyBB commented:
@JordanMussi Thanks, changes pushed to use links only. 

Strange, your probably right though :P

Ben-MyBB commented:
@euantorano Just with the various Captcha changes being made in .23, I wonder if we should include this relatively small PR?

euantorano commented:
Might make sense to do so, yeah. 

On Thu, 18 Jun 2020, at 20:05, Ben wrote:
> 
> 
> @euantorano <https://github.com/euantorano> Just with the various 
> Captcha changes being made in .23, I wonder if we should include this 
> relatively small PR?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/mybb/mybb/pull/4067#issuecomment-646251387>, or 
> unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AAFW24KWHAJI6DFNT33DURDRXJQRHANCNFSM4N4SXSMQ>.
>

JordanMussi commented:
Bump down lol

Ben-MyBB commented:
Same thing!
